crates/hir/src/semantics.rs

```diff
@@ -354,6 +356,10 @@ impl<'db, DB: HirDatabase> Semantics<'db, DB> {
         self.imp.resolve_bind_pat_to_const(pat)
     }
 
+    pub fn resolve_derive_ident(&self, ident: &ast::Ident) -> Option<PathResolution> {
```

Comment:
```suggestion
    pub fn resolve_derive_ident(&self, derive: &ast::Attr, ident: &ast::Ident) -> Option<PathResolution> {
        debug_assert!(ident.syntax().ancestors().any(|it| it == derive.syntax()));
```

The main benefit is that API becomes a touch more obvious, but it also allows us to avoid duplicate work as the caller probably already has attr.

Reply:
Ye I wasn't sure about this because now there is the (weak) requirement that `ident` is a child of `derive`. Though you are right, the duplicate work avoidance as well as making the API more obvious outweighs this.

Reply:
>now there is the (weak) requirement

Yeah, I see this as a benefit -- the API requires not an arbitrary ident, but an ident which comes from derive, which makes the API easier to understand. 

We *could* accept arbitrary `SytnaxNode`s for other semantics API, but being more strict about inputs is seems to be better  in this case.

Reply:
Ye that is a very good point.